pyproject.toml

```diff
@@ -190,3 +190,10 @@ norecursedirs = [
     "build",
 ]
 
+[dependency-groups]
+dev = [
+    "pydata-sphinx-theme>=0.14.4",
+    "sphinx-autobuild>=2021.3.14",
+    "sphinx-copybutton>=0.5.2",
+]
```

Comment:
@niccokunzmann I use uv for managing dependencies. I added these with `uv add pydata-sphinx-theme sphinx-autobuild sphinx-copybutton`, and it automatically ensures that the dependencies are compatible. I highly recommend using uv. See https://docs.astral.sh/uv/concepts/projects/dependencies/.

You also might consider managing test dependencies the same way, and merging them into this `dev` group. Developers should not have to keep separate groups, and it encourages them to write both tests and docs with a contribution.

Reply:
Hm. It is a bit duplicated at the moment.

If UV works, it would be nice to create a PR that documents how to use it.

Reply:
How do you currently manage dependencies? IOW, what tasks would you want to perform?

I could steal from the project I maintain.

https://plone-sphinx-theme.readthedocs.io/guides/contribute.html#update-requirements

I also use GNU Make to invoke uv and friends to make it easy to contribute.